### PR TITLE
[Ruby 2.5] Hash#transform_keys

### PR DESCRIPTION
**Pull request by andrykonchin**
https://bugs.ruby-lang.org/issues/13583

Specs are based on the specs for `Hash#transform_values`.
There are similar specs in `Rails` [repository](https://github.com/rails/rails/blob/5-1-stable/activesupport/test/core_ext/hash/transform_keys_test.rb). Not sure we should just copy-past them

### CODE REVIEW

**Comment by marcandre** [comment]
Again, I'm really sorry I didn't check your PR for this; not fun for you and more work for me lol!

Very good PR. I merged it.

I took the liberty from removing the "returns nil if we break from the block" for a couple of reasons:
- it doesn't return `nil`, it returns whatever value was given to `break` (of `nil` when none is given as in your example)
- this is not really relevant to `transform_keys/values`. For example, if you implement `transform_keys/values` in Ruby, it is simply not possible to not return the break value (short of raising an error within an ensure or something silly like that)
- we'd have to test that for all methods using blocks, and we'd also have to test with `throw`, `raise` and `return`. And `next` too.
In short, I feel that Rubyspec takes for granted that any implementation will have a consistent handling of blocks with regards to flow control.
